src/main/java/org/folio/service/file/S3JobRunningVerticle.java

```diff
@@ -0,0 +1,330 @@
+                      System.currentTimeMillis() - startTimeStamp,
+                      workersLeft
+                    );
+                  })
+              );
+
+              // do it one more time in hope that there more items in the queue
+              if (workersInUse.get() < maxWorkersCount) {
+                pollForJobs();
+              }
+            },
+            () -> LOGGER.info("No Items available to run.")
+          )
+        )
+        .onFailure(err -> LOGGER.error("Unable to get job from queue:", err));
+    }
+  }
+
+  protected Future<QueueJob> processQueueItem(DataImportQueueItem queueItem) {
+    LOGGER.info(
+      "Starting to process job execution {}",
+      queueItem.getJobExecutionId()
+    );
+
+    OkapiConnectionParams params = getConnectionParams(queueItem);
+
+    // we need to store out here to ensure it is properly deleted
+    // on failure and success
+    AtomicReference<File> localFile = new AtomicReference<>();
+
+    return Future
+      .succeededFuture(new QueueJob().withQueueItem(queueItem))
+      .compose((QueueJob job) ->
+        createLocalFile(queueItem)
+          .map((File file) -> {
+            localFile.set(file);
+            return job.withFile(file);
+          })
+      )
+      .compose(job ->
+        uploadDefinitionService
+          .getJobExecutionById(queueItem.getJobExecutionId(), params)
+          .map(job::withJobExecution)
+      )
+      .compose(job ->
+        updateJobExecutionStatusSafely(
+          job.getJobExecution().getId(),
+          new StatusDto().withStatus(StatusDto.Status.PROCESSING_IN_PROGRESS),
+          params
+        )
+          .map(v -> job)
+      )
+      .compose(this::downloadFromS3)
+      .compose(job ->
+        fileProcessor
+          .processFile(
+            job.getFile(),
+            job.getJobExecution().getId(),
+            // this is the only part used on our end
+            new JobProfileInfo()
+              .withDataType(
+                JobProfileInfo.DataType.fromValue(
+                  job.getQueueItem().getDataType()
+                )
+              ),
+            params
+          )
+          .map(v -> job)
+      )
+      .onFailure((Throwable err) -> {
+        LOGGER.error("Unable to start chunk {}", queueItem, err);
+
+        updateJobExecutionStatusSafely(
+          queueItem.getJobExecutionId(),
+          new StatusDto()
+            .withErrorStatus(ErrorStatus.FILE_PROCESSING_ERROR)
+            .withStatus(StatusDto.Status.ERROR),
+          params
+        );
+      })
+      .onSuccess((QueueJob result) ->
+        LOGGER.info(
+          "Completed processing job execution {}!",
+          queueItem.getJobExecutionId()
+        )
+      )
+      .onComplete((AsyncResult<QueueJob> v) -> {
+        queueItemDao.deleteDataImportQueueItem(queueItem.getId());
+
+        File file = localFile.get();
+        if (file != null) {
+          try {
+            FileUtils.delete(file);
```

Comment:
can vertx fileSystem instance be used for deletion here as well, just like it is used for local file creation?

Reply:
Just added this, thanks!